Assets/Scripts/HUD/Controller.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Player;
+using UnityEngine;
+
+namespace HUD
+{
+    public class Controller : MonoBehaviour
+    {
+        // !! Don't forget to attach a Player game object to the HUD !!
+        public GameObject Player;
+        public HealthBar HealthBar;
+        public Inventory Inventory;
+
+        //Used to access hitPoints of player
+        private Player.Controller _playerController;
+        //Used to access the weapons owned by player
+        private InventoryManager _playerInventoryManager;
+        
+        void Start()
+        {
+            _playerController = Player.GetComponent<Player.Controller>();
+            UpdateHealthBar();
+            SetUpHealthBarObservers();
+            
+            _playerInventoryManager = Player.GetComponent<InventoryManager>();
```

Comment:
Since the Player gameobject is tagged with "Player", and you only really need the components and not the gameobject itself, I would:
1. Get rid of the Player attribute in this class
2. To get the components, call `GameObject.FindWithTag("Player")` and get the components through that gameobject

Reply:
done